ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2

```diff
@@ -525,6 +525,21 @@ interface {{ ethernet_interface.name }}
    spanning-tree guard {{ ethernet_interface.spanning_tree_guard }}
 {%             endif %}
 {%         endif %}
+{%         if ethernet_interface.sflow.enabled is arista.avd.defined(true) %}
+   sflow enable
+{%         elif ethernet_interface.sflow.enabled is arista.avd.defined(false) %}
+   no sflow enable
+{%         endif %}
+{%         if ethernet_interface.sflow.egress.enabled is arista.avd.defined(true) %}
+   sflow egress enable
+{%         elif ethernet_interface.sflow.egress.enabled is arista.avd.defined(false) %}
+   no sflow egress enable
+{%         endif %}
+{%         if ethernet_interface.sflow.egress.unmodified_enabled is arista.avd.defined(true) %}
+   sflow egress unmodified enable
+{%         elif ethernet_interface.sflow.egress.unmodified_enabled is arista.avd.defined(false) %}
+   no sflow egress unmodified enable
+{%         endif %}
```

Comment:
To avoid parsing 6 `if`s for every interface, we can start by checking the outer sflow key - add similar for port-channel:
```suggestion
{%         if ethernet_interface.sflow is arista.avd.defined %}
{%             if ethernet_interface.sflow.enabled is arista.avd.defined(true) %}
   sflow enable
{%             elif ethernet_interface.sflow.enabled is arista.avd.defined(false) %}
   no sflow enable
{%             endif %}
{%             if ethernet_interface.sflow.egress.enabled is arista.avd.defined(true) %}
   sflow egress enable
{%             elif ethernet_interface.sflow.egress.enabled is arista.avd.defined(false) %}
   no sflow egress enable
{%             endif %}
{%             if ethernet_interface.sflow.egress.unmodified_enabled is arista.avd.defined(true) %}
   sflow egress unmodified enable
{%             elif ethernet_interface.sflow.egress.unmodified_enabled is arista.avd.defined(false) %}
   no sflow egress unmodified enable
{%             endif %}
{%         endif %}
```

Reply:
made requested changes

```j2
{%         if ethernet_interface.sflow is arista.avd.defined %}
{%             if ethernet_interface.sflow.enable is arista.avd.defined(true) %}
   sflow enable
{%             elif ethernet_interface.sflow.enable is arista.avd.defined(false) %}
   no sflow enable
{%             endif %}
{%             if ethernet_interface.sflow.egress.enable is arista.avd.defined(true) %}
   sflow egress enable
{%             elif ethernet_interface.sflow.egress.enable is arista.avd.defined(false) %}
   no sflow egress enable
{%             endif %}
{%             if ethernet_interface.sflow.egress.unmodified_enable is arista.avd.defined(true) %}
   sflow egress unmodified enable
{%             elif ethernet_interface.sflow.egress.unmodified_enable is arista.avd.defined(false) %}
   no sflow egress unmodified enable
{%             endif %}
{%         endif %}
```